cmd/encryption-v1.go

```diff
@@ -923,16 +923,18 @@ func getDecryptedETag(headers http.Header, objInfo ObjectInfo, copySource bool)
 	if crypto.IsMultiPart(objInfo.UserDefined) {
 		return objInfo.ETag
 	}
+
+	// If source is SSE-C encrypted and this is a SSE-C Copy operation
+	// we should return the last 32 bytes for the ETag, as there is no
+	// decryption necessary to comply with AWS S3 spec.
 	if crypto.SSECopy.IsRequested(headers) {
-		key, err = crypto.SSECopy.ParseHTTP(headers)
-		if err != nil {
-			return objInfo.ETag
-		}
+		return objInfo.ETag[len(objInfo.ETag)-32:]
```

Comment:
This change can be reverted since now CopyCondition check handles this appropriately.

Reply:
this change needs to be reverted